Mesh_3/test/Mesh_3/test_max_edge_distance.cpp

```diff
@@ -0,0 +1,183 @@
+
+#include <CGAL/Mesh_triangulation_3.h>
+#include <CGAL/Mesh_complex_3_in_triangulation_3.h>
+#include <CGAL/Mesh_criteria_3.h>
+#include <CGAL/Surface_mesh.h>
+
+#include <CGAL/Polyhedral_mesh_domain_with_features_3.h>
+#include <CGAL/Labeled_mesh_domain_3.h>
+#include <CGAL/Mesh_3/Detect_features_in_image.h>
+#include <CGAL/make_mesh_3.h>
+
+#include <CGAL/Sizing_field_with_aabb_tree.h>
+
+#include "test_meshing_utilities.h"
+
+#define CGAL_MESH_3_VERBOSE true
+
+template <typename Concurrency_tag = CGAL::Sequential_tag>
+struct Distance_polyhedral_tester : public Tester<K_e_i>
+{
+  typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
+
+  // Domain
+  typedef CGAL::Surface_mesh<K::Point_3> Polyhedron;
+  typedef CGAL::Polyhedral_mesh_domain_with_features_3<K, Polyhedron> Mesh_domain;
+
+  // Triangulation
+  typedef typename CGAL::Mesh_triangulation_3<Mesh_domain, K, Concurrency_tag>::type Tr;
+
+  typedef CGAL::Mesh_complex_3_in_triangulation_3<Tr> C3t3;
+
+  // Criteria
+  typedef CGAL::Mesh_criteria_3<Tr> Mesh_criteria;
+
+public:
+  void operator()(std::size_t expected_nb_vertices, std::size_t expected_nb_triangles)
+  {
+    const std::string fname = CGAL::data_file_path("meshes/dragknob.off");
+
+    std::ifstream input(fname);
+    using namespace CGAL::parameters;
+
+    Polyhedron polyhedron;
+    input >> polyhedron;
+    if (input.fail()) {
+      std::cerr << "Error: Cannot read file " << fname << std::endl;
+      return;
+    }
+
+    if (!CGAL::is_triangle_mesh(polyhedron)) {
+      std::cerr << "Input geometry is not triangulated." << std::endl;
+      return;
+    }
+
+    // Create domain
+    Mesh_domain domain(polyhedron);
+
+    domain.detect_features(40);
+
+    // Mesh criteria
+    Mesh_criteria criteria(edge_size = 0.074,
+        edge_distance = 0.00074,
+        facet_distance = 0.0074,
+        facet_angle = 25,
+        facet_size = 0.074,
+        cell_radius_edge_ratio = 3,
+        cell_size = 0.074);
+
+    // Mesh generation
+    C3t3 c3t3 = CGAL::make_mesh_3<C3t3>(domain, criteria, no_perturb(), no_exude());
+
+    std::size_t nb_vertices  = c3t3.triangulation().number_of_vertices();
+    std::size_t nb_triangles = c3t3.number_of_facets_in_complex();
+
+    this->verify_c3t3(c3t3, domain, Polyhedral_tag(), expected_nb_vertices * 0.95, expected_nb_vertices * 1.05, expected_nb_triangles * 0.95, expected_nb_triangles * 1.05);
+
+    // verify that there are more vertices than without criteria
+
+    Mesh_criteria criteria_without(edge_size = 0.074,
+        facet_distance = 0.0074,
+        facet_angle = 25,
+        facet_size = 0.074,
+        cell_radius_edge_ratio = 3,
+        cell_size = 0.074);
+
+    // Mesh generation
+    c3t3 = CGAL::make_mesh_3<C3t3>(domain, criteria_without, no_perturb(), no_exude());
+    assert(nb_vertices  > c3t3.triangulation().number_of_vertices());
+    assert(nb_triangles > c3t3.number_of_facets_in_complex());
```

Comment:
can we think of a more precise test?

Reply:
maybe check that generated feature edges are really closer than the required bound?

Reply:
up - think comment is still up to date